parts/linux/cloud-init/artifacts/reconcile-private-hosts.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o pipefail
+SLEEP_SECONDS=15
+clusterFQDN={{GetParameter "kubernetesEndpoint"}}
+if [[ $clusterFQDN != *.privatelink.* ]]; then
+  echo "skip reconcile hosts for $clusterFQDN since it's not AKS private cluster"
+  exit 0
+fi
+echo "clusterFQDN: $clusterFQDN"
+function get-apiserver-ip-from-tags() {
+  tags=$(curl -sSL -H "Metadata: true" "http://169.254.169.254/metadata/instance/compute/tags?api-version=2019-03-11&format=text")
+  if [ "$?" == "0" ]; then
+    IFS=";" read -ra tagList <<< "$tags"
+    for i in "${tagList[@]}"; do
+      tagKey=$(cut -d":" -f1 <<<$i)
+      tagValue=$(cut -d":" -f2 <<<$i)
+      if [ "$tagKey" == "aksAPIServerIPAddress" ]; then
```

Comment:
can we make this case-insensitive?
sample from stackoverflow https://stackoverflow.com/questions/15558101/case-insensitive-comparison-in-bash
while egrep -iq '^(y|yes)$' <<< $yn; do
  your_logic_here
done

Reply:
fixed to case-insensitive.